src/main/java/com/bankbazaar/test/controller/DepartmentController.java

```diff
@@ -0,0 +1,49 @@
+package com.bankbazaar.test.controller;
+
+import com.bankbazaar.test.entity.Department;
+import com.bankbazaar.test.service.DepartmentService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class DepartmentController {
+    @Autowired
+    private DepartmentService service;
+
+    @PostMapping("/addDepartment")
+    public Department addDepartment(@RequestBody Department department) {
+        return service.saveDepartment(department);
+    }
+
+    @PostMapping("/addDepartmentList")
+    public List<Department> addDepartmentList(@RequestBody List<Department> departments) {
+        return service.saveDepartmentList(departments);
+    }
+
+    @GetMapping("/department")
+    public List<Department> findAllDepartment() {
+        return service.getDepartmentList();
+    }
+
+    @GetMapping("/departmentById/{id}")
```

Comment:
This should return a 404 response when nothing is found. Implement it for other get apis as well.

Use ResponseEntity for the response

Reply:
Necessary changes have been made to implement ResponseEntity